code/graphics/generic.cpp

```diff
@@ -219,6 +224,29 @@ int generic_anim_stream(generic_anim *ga)
 
 		ga->previous_frame = -1;
 	}
+	else if (ga->type == BM_TYPE_PNG) {
+		if (ga->png.anim == nullptr) {
+			try {
+				ga->png.anim = new apng::apng_ani(ga->filename);
```

Comment:
unfortunately I ran into a cascade of changes in ship.cpp; where the generic_anim struct is copied around and it's non-trivial to fix it up all, it'll probably need a moderate rewrite of the struct and all the places it's used.  I may come back to this in future if I'm feeling keen.